tgui/packages/tgui/interfaces/KitchenMachine.js

```diff
@@ -0,0 +1,107 @@
+import { useBackend } from '../backend';
+import { Button, Section, Table, Flex, Icon, Dimmer } from '../components';
+import { Window } from '../layouts';
+
+export const KitchenMachine = (props, context) => {
+  const { data } = useBackend(context);
+  const { ingredients } = data;
+
+  return (
+    <Window resizable>
+      <Window.Content scrollable display="flex" className="Layout__content--flexColumn">
+        <Operating/>
+        <KitchenTop/>
+        <Section title="Ingredients">
+          <Table>
+            {ingredients.map((product) => (
+              <IngredientRow
+                key={product.name}
+                product={product}
+                productStock={product.units}
+              />
+            ))}
+          </Table>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const KitchenTop = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { inactive,
+          tooltip
+        } = data;
+
+  return (
+    <Section title="Controls">
+      <Flex>
+        <Flex.Item width="50%" mr="3px">
+          <Button
+            fluid
+            textAlign="center"
+            icon="power-off"
+            disabled={inactive}
+            tooltip={tooltip}
+            tooltipPosition="bottom"
+            content="Activate"
+            onClick={() => act('cook')}
+          />
+        </Flex.Item>
+        <Flex.Item width="50%" mr="3px">
+          <Button
+            fluid
+            textAlign="center"
+            icon="eject"
+            disabled={inactive}
+            tooltip={inactive ? tooltip : ""}
+            tooltipPosition="bottom"
+            content="Eject Contents"
+            onClick={() => act('eject')}
+          />
+        </Flex.Item>
+      </Flex>
+    </Section>
+  );
+}
+
+export const Operating = (props, context) => {
+  const { data } = useBackend(context);
+  const { operating,
+          name
+        } = data;
+
+  if (operating) {
+    return (
+      <Dimmer>
+        <Flex mb="30px">
+          <Flex.Item
+            bold
+            color="silver"
+            textAlign="center">
+            <Icon
+              name="spinner"
+              spin
+              size={4}
+              mb="15px"
+            /><br />
+            The {name} is processing...
+          </Flex.Item>
+        </Flex>
+      </Dimmer>
+    );
+  }
+};
+
+export const IngredientRow = (props, context) => {
```

Comment:
```suggestion
const IngredientRow = (props, context) => {
```
honestly, you probably don't even need to extract this chunk of UI into it's own const since you only ever use it once. The amount of code to abstract it actually takes up more space then it would to just put the table.row inline.

Reply:
Yeah, I kinda wanted to make it so the rows alternated colors, but I couldn't figure out how to do that, this is sorta a remnant of that. I'll move it back

Reply:
use SASS styling
```css
tr:nth-child(even) {
  background-color: Lightgreen;
}
```

Reply:
Yeah this is too difficult for me lol, maybe some day someone else can implement that

Reply:
all you need to do is define a new style class and pass it to the table element